### PR TITLE
Dataman Purchase Screens UI

### PR DESCRIPTION
**Pull request by sameera222**
- [x] Create purchase Screen  
- [x] Create Material Request Entry Screen
- [x] Create Place Order Screen
- [x] Create  Receipt Screen

### CODE REVIEW

File: apps/Dataman/package.json [hunk 1]
```diff
@@ -25,13 +27,13 @@
...
   },
   "devDependencies": {
-    "eslint-config-custom": "*",
     "@babel/core": "^7.19.3",
     "@babel/preset-env": "^7.1.6",
     "@types/react": "~18.0.25",
     "@types/react-dom": "~18.0.9",
     "@types/react-native": "~0.70.5",
     "babel-preset-expo": "~9.2.1",
+    "eslint-config-custom": "workspace:*",
```
Comment (sutyum) [inline apps/Dataman/package.json]: @chandantiwari1 was there an issue caused by this change?

File: apps/Dataman/src/screens/MaterialApproval.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          Deny
+        </Button>
+      </View>
+    </View>
+  );
+};
+
+export default MaterialApproval;
+
+const styles = StyleSheet.create({
```
Comment (sutyum) [inline apps/Dataman/src/screens/MaterialApproval.tsx]: why can't we use tailwind here

File: apps/Dataman/src/screens/MaterialRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          <Item arrow="horizontal">
+            <Item.Brief>
+              <Text style={styles.list_text}>Voucher Type</Text>
+              <AntDesign name="questioncircleo" />
+            </Item.Brief>
+            <Text style={styles.text}>Search Here</Text>
+          </Item>
+        </List>
+
+        <List>
```
Comment (sutyum) [inline apps/Dataman/src/screens/MaterialRequest.tsx:39]: refactor this into an array containing list data and use .map() to iteratate over this list in order to create the list view

File: apps/Dataman/src/screens/PlaceOrder.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+const PlaceOrder = () => {
+  const Item = List.Item;
+
+  return (
+    <View style={{ height: "100%" }}>
+      <Navbar />
+      <PageHeader text="Place Purchase Order" />
+      <ScrollView>
+        <List>
```
Comment (sutyum) [inline apps/Dataman/src/screens/PlaceOrder.tsx:20]: refactor this into an array containing list data and use .map() to iteratate over this list in order to create the list view

File: apps/Dataman/src/screens/Purchase.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          <Card.Body>
+            <Carousel
+              dotActiveStyle={{
+                backgroundColor: "#1677FF",
+                width: 23,
+                height: 7,
+              }}
+              style={{ height: 200, marginHorizontal: 10 }}
+            >
+              {[1, 2, 3, 4].map((item, index) => (
```
Comment (sutyum) [inline apps/Dataman/src/screens/Purchase.tsx]: refactor this list out

File: apps/Dataman/src/screens/Receipt.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+const Receipt = () => {
+  const Item = List.Item;
+  return (
+    <View style={{ height: "100%" }}>
+      <Navbar />
+      <PageHeader text="Good Receipt Entry" />
+      <ScrollView>
+        <List>
+          <Item arrow="horizontal">
```
Comment (sutyum) [inline apps/Dataman/src/screens/Receipt.tsx]: refactor this into an array containing list data and use .map() to iteratate over this list in order to create the list view

**Review by sutyum — CHANGES_REQUESTED** [review]

File: apps/Dataman/package.json (hunk 1 above)
Comment (chandantiwari1) [reply apps/Dataman/package.json]: No sir

File: apps/Dataman/src/screens/MaterialApproval.tsx (hunk 2 above)
Comment (chandantiwari1) [reply apps/Dataman/src/screens/MaterialApproval.tsx]: If tailwind css is not working then  use ///reference or add a global environment file which declares tailwind css as it is mentioned in their documentation
@sameera222

Comment (sameera222) [reply apps/Dataman/src/screens/MaterialApproval.tsx]: Tailwind does not work on the components of the ant design. If we want to use Tailwind, then we have to use React Native components. That's why I applied normal styling

File: apps/Dataman/src/screens/MaterialRequest.tsx (hunk 3 above)
Comment (sameera222) [reply apps/Dataman/src/screens/MaterialRequest.tsx:39]: Done

File: apps/Dataman/src/screens/PlaceOrder.tsx (hunk 4 above)
Comment (sameera222) [reply apps/Dataman/src/screens/PlaceOrder.tsx:20]: This is done

File: apps/Dataman/src/screens/Purchase.tsx (hunk 5 above)
Comment (sameera222) [reply apps/Dataman/src/screens/Purchase.tsx]: Done

File: apps/Dataman/src/screens/Receipt.tsx (hunk 6 above)
Comment (sameera222) [reply apps/Dataman/src/screens/Receipt.tsx]: Done

File: apps/Dataman/src/components/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { Button, View } from "@ant-design/react-native";
+
+export const FormButton = () => {
+  return (
+    <View style={{ alignItems: "center", marginVertical: 10 }}>
```
Comment (sutyum) [inline apps/Dataman/src/components/Button.tsx]: why not tailwind, same issue in various other places

File: apps/Dataman/src/components/TabBar.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  MaterialCommunityIcons,
+  Octicons,
+} from "@expo/vector-icons";
+
+export const Tab = () => {
+  const [selectedTab, setSelectedTab] = useState("");
+  const changeTab = (tabName: any) => {
+    setSelectedTab(tabName);
+  };
+
```
Comment (sutyum) [inline apps/Dataman/src/components/TabBar.tsx:15]: tailwind

File: apps/Dataman/src/screens/MaterialApproval.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+            Indent Number
+          </Text>
+          <InputField placeholder="Enter Number" />
+        </Item>
+      </List>
+
+      <View className=" flex flex-col items-center justify-center gap-1 mt-10">
+        <Button
+          type="primary"
+          style={{
```
Comment (sutyum) [inline apps/Dataman/src/screens/MaterialApproval.tsx]: why

**Review by sutyum — CHANGES_REQUESTED** [review]

**Review by sutyum — APPROVED** [review]

File: apps/Dataman/src/components/Button.tsx (hunk 7 above)
Comment (sameera222) [reply apps/Dataman/src/components/Button.tsx]: Issue resolved

File: apps/Dataman/src/components/TabBar.tsx (hunk 8 above)
Comment (sameera222) [reply apps/Dataman/src/components/TabBar.tsx:15]: Done

File: apps/Dataman/src/screens/MaterialApproval.tsx (hunk 9 above)
Comment (sameera222) [reply apps/Dataman/src/screens/MaterialApproval.tsx]: Issue resolved using tailwind css

**Pull request merged**